File: Sid's Levels/Level - 2/Strings/Longest Common Prefix.cpp
class Solution {
public:
    //OM GAN GANAPATHAYE NAMO NAMAH 
    //JAI SHRI RAM 
    //JAI BAJRANGBALI 
    //AMME NARAYANA, DEVI NARAYANA, LAKSHMI NARAYANA, BHADRE NARAYANA
    string helper(string s1, string s2)
    {
        int cntr = 0;
        string s;
        for(int i = 0; i < min(s1.length(), s2.length()); i++)
        {
            if(s1[i] != s2[i])
                break;
            cntr++;
        }
        return s1.substr(0, cntr);
    }
    string longestCommonPrefix(vector<string>& strs) {
        string res;
        if(strs.size() == 1)
            return strs[0];
        res = helper(strs[0], strs[1]);
        for(int i = 2; i < strs.size(); i++)
        {
            res = helper(res, strs[i]);
        }
        return res;
    }
};

File: DSA Revision/Sid/Strings/Longest Common Prefix.java
class Solution {
    //OM GAN GANAPATHAYE NAMO NAMAH 
    //JAI SHRI RAM 
    //JAI BAJRANGBALI 
    //AMME NARAYANA, DEVI NARAYANA, LAKSHMI NARAYANA, BHADRE NARAYANA
    public static StringBuilder lcp(String s1, String s2)
    {
        int n = s1.length();
        int m = s2.length();
        int i = 0, j = 0;
        while(i < n && j < m)
        {
            if(s1.charAt(i) == s2.charAt(j))
            {
                i++;
                j++;
            }
            else 
                break;
        }
        StringBuilder res = new StringBuilder();
        for(int ind = 0; ind < i; ind++)
        {
            res.append(s1.charAt(ind));
        }
        return res;
    }
    public String longestCommonPrefix(String[] strs) {
        StringBuilder lc = new StringBuilder(strs[0]);
        for(int i = 1; i < strs.length; i++)
        {
            lc = lcp(lc.toString(), strs[i]);
        }
        return lc.toString();
    }
}
